src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,20 +66,21 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
```

Comment:
flow is not smart enough to know that this check is flowing through the variable for the next condition